### PR TITLE
Fixed DatasetDao queries to return input and output facets

### PR DESCRIPTION
**Pull request by collado-mike**
I think this addresses https://github.com/MarquezProject/marquez/issues/1360 . Tested by spinning up the local docker image with seed data and ran the following bash
```

RUN_ID=60b858cc-faaf-498a-a242-e774f98a937a
curl -X POST http://localhost:5000/api/v1/namespaces/food_delivery/jobs/example.etl_delivery_7_days/runs -H 'Content-Type: application/json' -d @- <<EOL
{
    "id": "$RUN_ID"
}
EOL

###
# Start a job run
curl -X POST http://localhost:5000/api/v1/jobs/runs/$RUN_ID/start

###
# Send facet to OpenLineage API with START 
curl -X POST http://localhost:5000/api/v1/lineage -H 'Content-Type: application/json' -d @- <<EOL

{
    "eventType": "START",
    "eventTime": "2021-06-30T00:00:00Z",
    "run": {
        "runId": "$RUN_ID",
        "facets": {
            "startEventFacet": "2021-06-30T00:00:00Z"
        }
    },
    "job": {
        "namespace": "food_delivery",
        "name": "example.etl_delivery_7_days"
    },
    "inputs": [],
    "outputs": [{
        "namespace": "food_delivery",
        "name": "public.delivery_7_days",
        "facets": {
            "startEventFacet": "2021-06-30T00:00:00Z"
        }
    }],
    "producer": "https://github.com/kachontep/marquez-0.15.0/openlineage-facets-reproduce"
}
EOL


###
# Complete a job run
curl -X POST http://localhost:5000/api/v1/jobs/runs/$RUN_ID/complete

curl -X POST http://localhost:5000/api/v1/lineage -H 'Content-Type: application/json' -d @- <<EOL
{
    "eventType": "COMPLETE",
    "eventTime": "2021-06-30T00:00:00Z",
    "run": {
        "runId": "$RUN_ID",
        "facets": {
            "completeEventFacet": "2021-06-30T00:00:00Z"
        }
    },
    "job": {
        "namespace": "food_delivery",
        "name": "example.etl_delivery_7_days"
    },
    "inputs": [],
    "outputs": [{
        "namespace": "food_delivery",
        "name": "public.delivery_7_days",
        "facets": {
            "completeEventFacet": "2021-06-30T00:00:00Z"
        }
    }],
    "producer": "https://github.com/kachontep/marquez-0.15.0/openlineage-facets-reproduce"
}
EOL
```
Then read the dataset definition with
```
curl -s "http://localhost:5000/api/v1/namespaces/food_delivery/datasets/public.delivery_7_days" | jq | less
```

Initially, the `facets` field was empty, as reported in the issue. After the SQL change, both expected facets are present.

I also took the opportunity to test the query against a large dataset and updated the query to run faster. I also added/updated unit tests that should have caught this issue. Validated unit tests failed before the fix and pass after.

### CODE REVIEW

File: api/src/main/java/marquez/db/DatasetDao.java [hunk 1]
```diff
@@ -130,32 +147,51 @@ default void setFields(Dataset ds) {
   Optional<DatasetRow> getUuid(String namespaceName, String datasetName);
 
   @SqlQuery(
-      "SELECT d.*, dv.fields, sv.schema_location, facets, t.tags\n"
-          + "FROM datasets d\n"
-          + "INNER JOIN dataset_versions dv on d.current_version_uuid = dv.uuid\n"
-          + "LEFT OUTER JOIN stream_versions AS sv ON sv.dataset_version_uuid = dv.uuid\n"
-          + "LEFT OUTER JOIN (\n"
+      "WITH selected_datasets AS (\n"
```
Comment (wslulciuc) [inline api/src/main/java/marquez/db/DatasetDao.java:150]: _Minor_: I _think_ it'd be possible to factor out some of the common SQL into a constant for reuse?

**Review by wslulciuc — APPROVED** [review]
@collado-mike, thanks for fixing this bug and for adding a test suite for `DatasetDao`! 💯

File: api/src/main/java/marquez/db/DatasetDao.java (hunk 1 above)
Comment (collado-mike) [reply api/src/main/java/marquez/db/DatasetDao.java:150]: yeah, we need to start incorporating some strategy to reuse SQL- I felt this badly when I was implementing the jobVersion/datasetVersions in the Run model.

Here, the `selected_datasets` differs between the two queries- one specifies namespace and name, the other just specifies namespace and applies limit/offset. The rest of the SQL is the same until the end- the `findAll` query applies an `ORDER BY` clause at the end. It seems silly to me to extract the middle out and concatenate, so my preference is to leave it as is until we either start organizing reusable sql or refactoring tables so we don't need quite so much.

Comment (wslulciuc) [reply api/src/main/java/marquez/db/DatasetDao.java:150]: Makes sense! I'll open an issue to begin evaluating some of the current SQL and a plan to support reusability. But, for now, this is great 👍

File: api/src/main/java/marquez/db/DatasetDao.java [hunk 2]
```diff
@@ -130,32 +147,51 @@ default void setFields(Dataset ds) {
...
+          + "    SELECT d.*\n"
+          + "    FROM datasets d\n"
+          + "    WHERE d.namespace_name=:namespaceName\n"
+          + "    ORDER BY d.name\n"
+          + "    LIMIT 10 OFFSET 0\n"
```
Comment (dominiquetipton) [inline api/src/main/java/marquez/db/DatasetDao.java:155]: @collado-mike I know this is already merged, but is this supposed to be `LIMIT :limit OFFSET :offset` instead?

Comment (collado-mike) [reply api/src/main/java/marquez/db/DatasetDao.java:155]: 🤦🏽‍♂️

That's what I get for going back to my editor for one more test.

Comment (dominiquetipton) [reply api/src/main/java/marquez/db/DatasetDao.java:155]: We've all done it haha 😅  Speaking of tests, in mine the ORDER BY, LIMIT, AND OFFSET had to be at the end of the query in order for them to behave as expected (although I'm sure you noticed that too).

**Pull request merged**